### ISSUE TITLE
bug(mat-stepper): a11y disabled state

### ISSUE THREAD

**Issue by csutorasr** [angular/components#23088]
#### Reproduction

Steps to reproduce:
1. Turn on VoiceOver or any screenreader.
2. Go to a linear stepper element with more non-optional steps.
3. Hover the second step in the stepper.
 
#### Expected Behavior

The disabled state should be announced. `aria-disabled` could solve this.

#### Actual Behavior

There is no information on disabled state.

#### Environment

- Angular: 12
- CDK/Material: 12
- Browser(s): Any
- Operating System (e.g. Windows, macOS, Ubuntu): Any
